components/collector/tests/base_collectors/test_collector.py

```diff
@@ -62,6 +63,8 @@ async def _fetch_measurements(self, mock_async_get_request, number=1, side_effec
             async with aiohttp.ClientSession() as session:
                 for _ in range(number):
                     await self.collector.collect_metrics(session)
+                    # Wait for the created tasks to finish
+                    await asyncio.gather(*asyncio.all_tasks() - {asyncio.current_task()})
```

Comment:
This looks magically cool, I guess it's fine within test code but you break it you buy it :D